cider/exec/nextgen/operators/SourceNode.h

```diff
@@ -30,13 +30,46 @@ namespace cider::exec::nextgen::operators {
 
 class SourceNode : public OpNode {
  public:
-  SourceNode(const ExprPtrVector& col_exprs);
+  template <typename T, IsVecOf<T, ExprPtr> = true>
+  SourceNode(T&& exprs) : exprs_(std::forward<T>(exprs)) {}
 
-  ExprPtrVector getExprs() override { return input_cols_; }
+  template <typename... T>
+  SourceNode(T&&... exprs) {
+    (exprs_.emplace_back(std::forward<T>(exprs)), ...);
+  }
+
+  ExprPtrVector getExprs() override { return exprs_; }
 
  private:
-  ExprPtrVector input_cols_;
+  ExprPtrVector exprs_;
 };
+
+class SourceTranslator : public Translator {
+ public:
+  template <typename T>
+  SourceTranslator(T&& exprs, std::unique_ptr<Translator> succ) {
+    node_ = SourceNode(std::forward<T>(exprs));
+    successor_.swap(succ);
+  }
+
+  template <typename... T>
+  SourceTranslator(T&&... exprs, std::unique_ptr<Translator> successor) {
+    node_ = SourceNode(std::forward<T>(exprs)...);
+    successor_.swap(successor);
+  }
+
+  SourceTranslator(SourceNode&& node, std::unique_ptr<Translator>&& succ)
+      : node_(std::move(node)), successor_(std::move(succ)) {}
+
+  void consume(Context& context) override;
+
+ private:
+  void codegen(Context& context);
+
+  SourceNode node_;
```

Comment:
you should add as much as comments to data fields.

Reply:
Add comments for field is a valid comment. Let's try to add them in the first place.